neuron.schema.json

```diff
@@ -0,0 +1,148 @@
+{
+    "$schema": "https://json-schema.org/draft/2019-09/schema",
+    "$id": "https://raw.githubusercontent.com/GenGeNN/GenGeNN-resources/master/neuron.json.schema",
+    "title": "Neuron",
+    "description": "A single neuron definition for GeNN JSON",
+    "type": "object",
+    "properties": {
+        "friendlyName": {
+            "description": "A friendly name for the neuron class",
+            "type": "string"
+        },
+        "summaryDescription": {
+            "description": "A description of the function of the neuron",
+            "type": "string"
+        },
+        "className": {
+            "description": "The programmatic name for the class",
+            "type": "string"
+        },
+        "parameters": {
+            "description": "All the parameter names for the simulation",
+            "type": "array",
+            "items": {
+                "type": "string"
+            }
+        },
+        "variables": {
+            "description": "All the variable names and types for the simulation",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "type": {
+                        "type": "string",
+                        "enum": ["scalar","vector"]
+                    }
+                },
+                "required": ["name", "type"],
+                "additionalProperties": false
+            }
+        },
+        "derivedParameters": {
+            "description": "All the names and derivation functions of secondary parameters in the simulation",
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    },
+                    "derivation": {
+                        "type": "object",
+                        "minProperties": 1,
+                        "properties": {
+                            "generic": {"type": "string", "minlength": 1},
+                            "python": {"type": "string", "minlength": 1},
+                            "cpp": {"type": "string", "minlength": 1}
+                        },
+                        "additionalProperties": {"type": "string", "minlength": 1}
+                    }
+                },
+                "required": ["name", "derivation"],
+                "additionalProperties": false
+            }
+        },
+        "additionalParameters": {
```

Comment:
I am not sure what would go here (but could just be me ...)?

Reply:
that was for the variable known as _extra_global_params_ in Python, and thought it was a more logical format - but could change if having different names if confusing